### ISSUE TITLE
asked for iTunes password repeatedly

### ISSUE THREAD

**Issue by squillopmedia** [Vitaa/IAPurchaseManager#7]
Hi,

I'm very grateful for this code, it does make life much easier.  Thank you !

but i see a weird thing when trying this out in my app (which is beta sandbox)

when buying the purchase, users are asked to enter password, then touch ID, then password again before the purchase is done.

is there something i am doing wrong ?  surely users should only see  the password requirement once per session ?

thanks for your help and your code !!

**Comment by Vitaa** [comment]
Hi! Thanks for using it!

hm, never seen this problem. Possible reasons:
1) maybe you call restoreCompletedTransactions every time you call purchaseProductWithId? 
2) is purchaseProductWithId called only once?

Let me know if you still have this issue

**Comment by squillopmedia** [comment]
hi, thanks for your answer.
I think this could be a sandbox problem.  I found a posting about it on Apple developer forums.. 

I have checked my code and I don't think i am doing either of the things you mention above.  I call each line only once. 

I see different results now on my test devices, sometimes it works fine - asks for password only once and purchase completes.  Other times, seems to be if i use touch ID, i am asked to touch 3 or more times and then the purchase fails !   If i try again, it might do same, or might work. 

Hard to track down an error like this !!!  But - my hope is that the sandbox beta TestFlight environment is what is causing it  - and when my app is live this won't happen !!

thanks again for your help and nice code !
